test/test_analyse.ml

```diff
@@ -111,4 +111,48 @@ let test_duniverse =
   in
   expect_test "duniverse" ~project ~expected
 
-let tests = [ test_simple; test_multiple_opam; test_duniverse ]
+let test_ocamlformat_vendored =
+  let project =
+    let open Gen_project in
+    [
+      File ("example.opam", opam);
+      (* This file is not parsed if ocamlformat is vendored *)
+      File (".ocamlformat", empty_file);
+      Folder
+        ( "duniverse",
+          [ Folder ("ocamlformat", [ File ("ocamlformat.opam", opam) ]) ] );
+    ]
+  in
+  let expected =
+    let open Analysis in
+    {
+      opam_files = [ "example.opam"; "duniverse/ocamlformat/ocamlformat.opam" ];
+      is_duniverse = true;
+      ocamlformat_version = Some Vendored;
+    }
+  in
+  expect_test "ocamlformat_vendored" ~project ~expected
+
+let test_ocamlformat_self =
+  let project =
+    let open Gen_project in
+    [ File ("ocamlformat.opam", opam); File (".ocamlformat", empty_file) ]
```

Comment:
Have you tested this on OCamlformat itself? I remember there being other blocking problems with that (such as OCamlformat using different configurations for subtrees of its source).

Reply:
That doesn't work on ocamlformat yet. Ocamlformat sources can be fixed so that `dune build @fmt` works.